src/Plugin.php

```diff
@@ -46,6 +46,29 @@ public static function init() {
     add_filter('post_gallery', __CLASS__ . '::post_gallery', 10, 2);
     add_action('print_media_templates', __CLASS__ . '::print_media_templates');
 
+    // Adds woocommerce specific settings (product thumbnail slider settings).
+    if (static::isPluginActive('woocommerce/woocommerce.php')) {
+      add_filter('woocommerce_get_settings_gallerya', __NAMESPACE__ . '\WooCommerce::woocommerce_get_settings_gallerya');
+    }
+
+    // Adds CSS class to single product image galleries to enable thumbnail slider via JS.
+    if (static::isPluginActive('woocommerce/woocommerce.php') && get_option('_' . Plugin::L10N . '_product_thumbnail_slider_enabled') === 'yes' && get_option('_' . Plugin::L10N . '_product_thumbnail_slider_bullet_nav_enabled') !== 'yes') {
+      add_filter('woocommerce_single_product_image_gallery_classes', function ($classes) {
+        if (in_array('woocommerce-product-gallery--with-images', $classes)) {
+          $classes[] = 'js-gallerya-product-thumbnail-slider';
+        }
+        return $classes;
+      });
+    }
+
+    // Exchanges thumbnails with bullets as control nav for the single product galleries.
+    if (static::isPluginActive('woocommerce/woocommerce.php') && get_option('_' . Plugin::L10N . '_product_thumbnail_slider_bullet_nav_enabled') === 'yes') {
+      add_filter('woocommerce_single_product_carousel_options', function ($args) {
+        $args['controlNav'] = TRUE;
+        return $args;
+      });
+    }
```

Comment:
IMO this could be refactored to make it more legible:
```php
// Adds CSS class to single product image galleries to enable thumbnail slider via JS.
if (static::isPluginActive('woocommerce/woocommerce.php') {
  add_filter('woocommerce_get_settings_gallerya', __NAMESPACE__ . '\WooCommerce::woocommerce_get_settings_gallerya');

  $slider_bullet_nav_enabled = get_option('_' . Plugin::L10N . '_product_thumbnail_slider_bullet_nav_enabled');

  if (get_option('_' . Plugin::L10N . '_product_thumbnail_slider_enabled') === 'yes' && $slider_bullet_nav_enabled !== 'yes') {
    add_filter('woocommerce_single_product_image_gallery_classes', __NAMESPACE__ . '\WooCommerce::woocommerce_single_product_image_gallery_classes');
  }

  // Exchanges thumbnails with bullets as control nav for the single product galleries.
  if ($slider_bullet_nav_enabled === 'yes') {
    add_filter('woocommerce_single_product_carousel_options', __NAMESPACE__ . '\WooCommerce::woocommerce_single_product_carousel_options');
  }
}
```
And then extract filters methods `woocommerce_single_product_image_gallery_classes` and `woocommerce_single_product_carousel_options` into class `WooCommerce`.

Reply:
Thanks, @juanlopez4691, this is much more legible! I've just refactored it.